submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/model/UserData.kt

```diff
@@ -0,0 +1,35 @@
+package ac.uk.ebi.biostd.persistence.model
+
+import java.io.Serializable
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.IdClass
+import javax.persistence.Table
+
+@Table(name = "UserData")
+@Entity
+@IdClass(UserDataId::class)
+class UserData(
+    @Id
+    val userId: Long,
+
+    @Id
+    val key: String,
+
+    var data: String
+)
+
+class UserDataId : Serializable {
+    var userId: Long = 0
+
+    @Column(name = "dataKey")
+    lateinit var key: String
+
+    constructor()
```

Comment:
Why do we need this?

Reply:
jpa requires an argument constructor for ID types

Reply:
got it